system-helper/flatpak-system-helper.c

```diff
@@ -190,6 +420,341 @@ get_sender_pid (GDBusMethodInvocation *invocation)
+      if (check_partial_pull_availability (repo_tmp, parts[1], commit, &out_location))
+        {
+          g_autoptr(GFile) user_dir_file = NULL;
+
+          g_debug ("Found a partial pull for ref %s at: %s", parts[1], out_location);
+          root_dir = g_path_get_dirname (out_location);
+          user_dir = g_strdup_printf ("%s-bindfs", root_dir);
+          user_dir_file = g_file_new_for_path (user_dir);
+          if (!g_file_query_exists (user_dir_file, NULL))
+            g_mkdir_with_parents (user_dir, 0755);
+          child_repo_path = g_steal_pointer (&out_location);
+        }
+      else
+        {
+          g_autoptr(OstreeRepo) child_repo = NULL;
+          g_auto(GLnxLockFile) child_repo_lock = { 0, };
+
+          g_autofree gchar *cache_dir_basename = g_strdup_printf ("flatpak-cache-%s-XXXXXX", parts[1]);
+          root_dir = g_mkdtemp_full (g_build_filename (repo_tmp, cache_dir_basename, NULL), 0755);
+          user_dir = g_strdup_printf ("%s-bindfs", root_dir);
+          if (g_mkdir_with_parents (user_dir, 0755) == -1)
+            {
+              glnx_throw_errno_prefix (&error, "Can't create directory for bindfs mount at %s", user_dir);
+              g_dbus_method_invocation_return_gerror (invocation, error);
+              return TRUE;
+            }
+
+          g_autoptr(GFile) root_dir_file = g_file_new_for_path (root_dir);
+          child_repo = flatpak_dir_create_child_repo (system, root_dir_file, &child_repo_lock, NULL, &error);
+          if (child_repo == NULL)
+            {
+              g_dbus_method_invocation_return_error (invocation, G_DBUS_ERROR, G_DBUS_ERROR_FAILED,
+                                                     "Failed to create OSTree child repo at %s", root_dir);
+              return TRUE;
+            }
+          GFile *repo_file = ostree_repo_get_path (child_repo);
+          child_repo_path = g_file_get_path (repo_file);
+        }
+
+      g_assert (child_repo_path != NULL);
+
+      new_pull = system_helper_ongoing_pull_new (object, invocation, ref, commit,
+                                                 uid, user_dir, root_dir, child_repo_path);
+
+      /* In bindfs, --force-user forces file-ownership to be seen in the mounted mirrored directory(i.e. user_dir),
+         while --create-for-user enforces the actual owner of the file during file creation on disk.  */
+      arg_user = g_strdup_printf ("--force-user=%"G_GUINT32_FORMAT, uid);
+      new_pull->bindfs = g_subprocess_new (G_SUBPROCESS_FLAGS_NONE,
+                                           &error,
+                                           bindfs_path,
+                                           arg_user,"-f",
+                                           "-o", "nonempty",
+                                           "--create-for-user=root",
+                                           "--create-for-group=root",
+                                           root_dir, user_dir, NULL);
```

Comment:
pass a GError here ?

Reply:
In `g_subprocess_new`, the `error` is passed as the second parameter. It's passed here but on the error path, there is a creation of our own error message  if G_DBUS_ERROR_SPAWN_FAILED. 
I'll change that to use this GError.

Reply:
ah, ok. I was led astray by the final NULL